migration/migration.go

```diff
@@ -0,0 +1,68 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/NethermindEth/juno/db"
+)
+
+type revision func(db.DB) error
+
+var revisions = []revision{
+	revision0000,
+}
+
+func MigrateIfNeeded(targetDB db.DB) error {
+	version, err := SchemaVersion(targetDB)
+	if err != nil {
+		return err
+	}
+
+	for i := version; i < uint64(len(revisions)); i++ {
+		if err = revisions[i](targetDB); err != nil {
+			return err
+		}
+
+		// revision returned with no errors, bump the version
+		if err = targetDB.Update(func(txn db.Transaction) error {
```

Comment:
What happens if the schema update is successful, but the process dies before updating the version here?

Reply:
Yeap, that is an edge case that we dont want to be left in. I will make sure everything happens within a transaction context.

Reply:
done